Lib/test/test_socket.py

```diff
@@ -165,6 +166,16 @@ def socket_setdefaulttimeout(timeout):
         socket.setdefaulttimeout(old_timeout)
 
 
+@contextlib.contextmanager
+def catch_malformed_data_warning(quiet=True):
```

Comment:
Does it catch them or ignore them? What is the behavior of quiet=False?

Reply:
Yes https://github.com/python/cpython/blob/8f324b7ecd2df3036fab098c4c8ac185ac07b277/Lib/test/support/warnings_helper.py#L152-L188
1. It catches warnings with `warnings.catch_warnings(record=True)`
2. Then it filters out ones that matched `("received malformed or improperly-truncated ancillary data", RuntimeWarning)`
3. If there are any left, the test fails
4. `quiet=True` handles the cases where no warning was raised - just by ignoring the last check, if `quiet=False` we are required to have at least one matched warning